passes/sat/qbfsat.cc

```diff
@@ -91,7 +98,10 @@ void recover_solution(QbfSolutionType &sol) {
 			log_assert(YS_REGEX_NS::regex_search(loc, hole_loc_regex));
 			log_assert(YS_REGEX_NS::regex_search(val, hole_val_regex));
 #endif
-			sol.hole_to_value[loc] = val;
+			RTLIL::AttrObject tmp;
+			tmp.set_src_attribute(loc);
+			pool<std::string> loc_pool = tmp.get_strpool_attribute(ID::src);
+			sol.hole_to_value[loc_pool] = val;
```

Comment:
Shouldn't that be identical to `auto locs =  split_tokens(loc, "|"); pool<std::string> loc_pool(locs.first, locs.second); ...`? No need to create a temp AttrObject.

Reply:
Yes, that should work fine, too.  This way, though, if the storage format of a strpool attribute changes (maybe there is a need to store `|` in some strpool attribute), this code will not need updating.  I don't really mind either way, so I'll change it if you prefer the `split_tokens()` approach.

Reply:
> This way, though, if the storage format of a strpool attribute changes (maybe there is a need to store `|` in some strpool attribute)

We'll have to escape it for a rigorous solution. On Linux the only thing that isn't valid in a path is `\0`, but using that as a separator is just asking for trouble.

Reply:
OK, I changed it.